src/main/java/com/homeadvisor/kafdrop/service/KafkaHighLevelConsumer.java

```diff
@@ -0,0 +1,157 @@
+package com.homeadvisor.kafdrop.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.Maps;
+import com.homeadvisor.kafdrop.config.KafkaConfiguration;
+import com.homeadvisor.kafdrop.model.TopicPartitionVO;
+import com.homeadvisor.kafdrop.model.TopicVO;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.Node;
+import org.apache.kafka.common.PartitionInfo;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * Created by Satendra Sahu on 9/20/18
+ */
+@Service
+public class KafkaHighLevelConsumer
+{
+    private final Logger LOG = LoggerFactory.getLogger(getClass());
+    @Autowired
+     private ObjectMapper objectMapper;
+    private KafkaConsumer<String, String> kafkaConsumer;
+
+    @Autowired
+    private KafkaConfiguration kafkaConfiguration;
+
+    public KafkaHighLevelConsumer() {}
+
+    @PostConstruct
+    private void initializeClient()
+    {
+        if (kafkaConsumer == null) {
+
+            Properties properties = new Properties();
+            properties.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, false);
+            properties.put(ConsumerConfig.GROUP_ID_CONFIG, "kafka-drop-consumer-group");
+            properties.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, kafkaConfiguration.getKeyDeserializer());
+            properties.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, kafkaConfiguration.getValueDeserializer());
+            properties.put(ConsumerConfig.MAX_POLL_RECORDS_CONFIG, 100);
+            properties.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest");
+            properties.put(ConsumerConfig.CLIENT_ID_CONFIG, "kafka-drop-client");
+            properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaConfiguration.getBrokerConnect());
+
+            if (kafkaConfiguration.getIsSecured() == true) {
+                properties.put("sasl.mechanism", "PLAIN");
+                properties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_PLAINTEXT");
```

Comment:
Will this restrict the ability to use other sasl.mechanism or alternative jaas.conf configurations?  Shouldn't it at least pull in the same sasl.mechanism as the config.
In fact, shouldn't it just pull in the whole config from jaas.conf?  Looking at how yahoo's kafka-manager does it there's still some code around it but that way there's no extra kafdrop config (alleviates the isSecured part) and push that back down to the underlying libraries (either kafka or jaas depending on which part we're talking about)